frontend/internal-packages/agent/src/deepModeling.ts

```diff
@@ -1,145 +1,43 @@
-import { AIMessage, HumanMessage } from '@langchain/core/messages'
-import { RunCollectorCallbackHandler } from '@langchain/core/tracers/run_collector'
-import type { Schema } from '@liam-hq/db-structure'
-import type { Result } from 'neverthrow'
-import { err, ok } from 'neverthrow'
-import { v4 as uuidv4 } from 'uuid'
-import { WORKFLOW_ERROR_MESSAGES } from './chat/workflow/constants'
-import { finalizeArtifactsNode } from './chat/workflow/nodes'
+import { err } from 'neverthrow'
 import { DEFAULT_RECURSION_LIMIT } from './chat/workflow/shared/langGraphUtils'
-import type { WorkflowConfigurable, WorkflowState } from './chat/workflow/types'
-import { withTimelineItemSync } from './chat/workflow/utils/withTimelineItemSync'
+import type { WorkflowConfigurable } from './chat/workflow/types'
 import { createGraph } from './createGraph'
-
-export type DeepModelingParams = {
-  userInput: string
-  schemaData: Schema
-  history: [string, string][]
-  organizationId: string
-  buildingSchemaId: string
-  latestVersionNumber: number
-  designSessionId: string
-  userId: string
-  recursionLimit?: number
-}
-
-export type DeepModelingResult = Result<WorkflowState, Error>
+import {
+  executeWorkflowWithTracking,
+  setupWorkflowState,
+} from './shared/workflowSetup'
+import type { AgentWorkflowParams, AgentWorkflowResult } from './types'
 
 /**
  * Execute Deep Modeling workflow
  */
 export const deepModeling = async (
-  params: DeepModelingParams,
+  params: AgentWorkflowParams,
   config: {
     configurable: WorkflowConfigurable
   },
-): Promise<DeepModelingResult> => {
-  const {
-    userInput,
-    schemaData,
-    history,
-    organizationId,
-    buildingSchemaId,
-    latestVersionNumber = 0,
-    designSessionId,
-    userId,
-    recursionLimit = DEFAULT_RECURSION_LIMIT,
-  } = params
-
-  const { repositories } = config.configurable
+): Promise<AgentWorkflowResult> => {
+  const { recursionLimit = DEFAULT_RECURSION_LIMIT } = params
 
-  // Convert history to BaseMessage objects
-  const messages = history.map(([role, content]) => {
-    return role === 'assistant'
-      ? new AIMessage(content)
-      : new HumanMessage(content)
-  })
+  // Setup workflow state with message conversion and timeline sync
+  const setupResult = await setupWorkflowState(params, config)
```

Comment:
Since I want to use the setup process with invokeDbAgent, I cut it out into a function.